### PR TITLE
AMDlibm: Fix build system after Spack env changes

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
Patch for existing versions to fix build system after Spack env changes.

### CODE REVIEW

File: var/spack/repos/builtin/packages/amdlibm/package.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ class Amdlibm(SConsPackage):
 
     patch("0001-libm-ose-Scripts-cleanup-pyc-files.patch", when="@2.2")
     patch("0002-libm-ose-prevent-log-v3.c-from-building.patch", when="@2.2")
+    patch("libm-ose-SconsSpack.patch", when="@3.1:4.2")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/amdlibm/package.py:64]: Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed
3. Where it was taken from

?

Comment (alalazo) [reply var/spack/repos/builtin/packages/amdlibm/package.py:64]: Also, just for my understanding, why the build system needs to be this tightly coupled to Spack (to the point it must know which env variables are set)?

Comment (trws) [reply var/spack/repos/builtin/packages/amdlibm/package.py:64]: It's scons, which clears the environment by default. 🤦 

Really wish we had a decent solution to that, so spack could just tell a spack-managed scons run to preserve its env vars.

**Review by trws — DISMISSED** [review]

File: var/spack/repos/builtin/packages/amdlibm/package.py (hunk 1 above)
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/amdlibm/package.py:64]: 1. What is the effect of the patch?

- Propagates a needed env var back into SCONS which deletes them all by default

2. Why it is needed ?

- It's the only way we see to make the build system work

3. Where it was taken from

- Backport of an internal patch to released versions

4. why the build system needs to be this tightly coupled to Spack (to the point it must know which env variables are set)?

- Per @trws - we don't want it to be, but we have no choice but to repropagate the required env vars.

Comment (alalazo) [reply var/spack/repos/builtin/packages/amdlibm/package.py:64]: @amd-toolchain-support Thanks for the explanation, but:

> Can you add **a brief comment above the patch directive** explaining:

can you write a comment above the patch directive? It helps when reading packages.

**Review by alalazo — APPROVED** [review]

**Pull request merged**